CRM/Financial/BAO/ExportFormat/DataProvider/Sage50CSVProvider.php

```diff
@@ -90,9 +90,14 @@ public static function runExportQuery($batchId) {
              LEFT JOIN civicrm_financial_item fi ON fi.id = efti.entity_id
              LEFT JOIN civicrm_financial_item fii ON (fii.id = (SELECT eftii.entity_id
                                                     FROM civicrm_entity_financial_trxn eftii
+                                                      INNER JOIN civicrm_financial_item fiii ON fiii.id = eftii.entity_id
```

Comment:
So given you will loop through the result of this query below to format the data, you can maybe remove some of unnecessary joins here and replace their values when formatting them, for example the joins you are doing with:

- civicrm_option_group 
- civicrm_option_value 
- civicrm_financial_account 
- civicrm_financial_type 

can be removed, and before looping through the result to format them, fetch the financial accounts, financial types and relevant option group values and replace them.

Reply:
@omarabuhussein  Good idea! Especially civicrm_financial_type and civicrm_financial_account and payment method.  I do not think we have time and budget to tweak this now, and these joins have been with CiviCRM core Journal CSV format.  When we hit the performance issue, then we could take it to consider improving in the future.

Reply:
👍